### PR TITLE
Add qir to azurebackend

### PR DESCRIPTION
**Pull request by Roland-djee**
This PR addresses the addition of submitting jobs as QIR bitcode files to through the `AzureBackend` instead of a Q# object.

Since Q# is likely to be removed from the submission process, this extension might be considered for renaming to `pytket-azure`. This could take place when it is moved to its own repo.

### CODE REVIEW

**Comment by cqc-alec** [comment]
Sorry I have only just seen this. Could you please resolve the merge conflict?

File: modules/pytket-qsharp/pytket/extensions/qsharp/backends/azure_quantum.py [hunk 1]
```diff
@@ -34,7 +33,8 @@
 
 from pytket.extensions.qsharp.backends.config import QSharpConfig
 from pytket.extensions.qsharp.backends.common import _QsharpBaseBackend
-from pytket.extensions.qsharp.qsharp_convert import tk_to_qsharp
```
Comment (cqc-alec) [inline modules/pytket-qsharp/pytket/extensions/qsharp/backends/azure_quantum.py:37]: We should get rid of `tk_to_qsharp` altogether if possible, but I see it is still used in the `QSharpBaseBackend`. Can we make that use QIR too?

**Comment by Roland-djee** [comment]
> Sorry I have only just seen this. Could you please resolve the merge conflict?

DOne. Though I would be a bit wary about it. I am not sure Azure fully support QIR submissions yet. There are ongoing discussions with MS people atm but I am not sure what the status of this is. Although the test might succeed, the results might be meaningless.

**Comment by cqc-alec** [comment]
> > Sorry I have only just seen this. Could you please resolve the merge conflict?
> 
> Done. Though I would be a bit wary about it. I am not sure Azure fully support QIR submissions yet. There are ongoing discussions with MS people atm but I am not sure what the status of this is. Although the test might succeed, the results might be meaningless.

OK. At least for noiseless simulator backends it should be possible to test if the results are meaningful.

**Comment by cqc-melf** [comment]
Is this something you want to get moved over to the new repo? https://github.com/CQCL/pytket-qiskit/

**Pull request closed without merging**